python-exercises/assignment/parking_lot.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+
+#This is  parking lot assignment code file
+
+import sys
+
+from prettytable import PrettyTable
+
+class Car:
+    """
+    Class of Car 
+    """
+    def __init__(self, registration_number, color):
+        # car registration number
+        self.registration_number = registration_number
+        # car color
+        self.color = color
+
+class Slot:
+    """
+    Class of Slot
+    """
+    def __init__(self, id):
+        # slot id
+        self.id = id
+        # car in current slot
+        self.car = None
+
+    def park_car(self, car: Car):
+        """
+        This function is to park a car in a slot
+        """
+        status = ''
+        if self.car is None:
+            self.car = car
+            status = 'OK'
+        else:
+            status = 'This slot is not empty!'
+        return status
+    
+    def get_registration_number(self):
+        """
+        This function is to get registration number of the car in the current slot
+        """
+        if self.car is not None:
+            return str(self.car.registration_number)
+        else:
+            return ('No car in this slot!')
+    
+    def get_color(self):
+        """
+        This function is to get color of the car in the current slot
+        """
+        if self.car is not None:
+            return str(self.car.color)
+        else:
+            return ('No car in this slot!')
+
+class ParkingLot:
+    """
+    Class of Parking Lot 
+    """
+    def __init__(self):
+        #parking lot size
+        self.size = 0
+        #current car number in the parking lot
+        self.car_count = 0
+        #Slot array
+        self.parking_lot_list = []
+        #Create status table
+        self.table = PrettyTable()
+
+    def create_parking_lot(self, capacity):
+        """
+        This function is to create a parking lot with specific slots
+        """
+        if capacity % 2 != 0 or capacity <= 0:
+            return 'No of slot must be even positive!'
+        self.size = capacity
+        for i in range(self.size):
+            self.parking_lot_list.append(Slot(i + 1))
+        return 'OK'
+        
+    def park(self, registration_number, color):
+        """
+        This function is to park a car in current slot
+        """
+        if self.car_count == self.size:
+            return 'Parking slot is full!'
+        else:
+            for slot in self.parking_lot_list:
+                if slot.car is None:
+                    slot.park_car(Car(registration_number, color))
+                    self.car_count += 1
+                    break
+            return 'OK'
+
+    def leave(self, slot):
```

Comment:
Would `slot_id` be relevant in this case? Assume that I'm a driver, I might know the slot ID but not the slot index 😉